app/services/user_invite_service.rb

```diff
@@ -11,7 +11,7 @@ def self.invite(email:, resource:, name: nil, roles: [], force: false)
     user = User.find_by(email: email)
 
     # return if user already has all the roles we're trying to add
-    if !force && user && roles.all? { |role| user.has_role?(role, resource) }
+    if !force && user && roles.all? { |role| user.has_role?(role, resource) && resource.email != user.email }
```

Comment:
@awwaiid dear me -- think this through again, also maybe it should be outside of the block instead of inside

Reply:
Hey @awwaiid , 
We are getting bug requests for this issue. If this PR is stale. I want to work on it.

Reply:
Yes, please move this out of the all? block and also check below for the `if user` block and think through what the behavior should be in the case the user exists but we are changing the email address